src/main/java/io/cdap/plugin/gcp/gcs/ServiceAccountAccessTokenProvider.java

```diff
@@ -43,19 +51,71 @@ public class ServiceAccountAccessTokenProvider implements AccessTokenProvider {
   private Configuration conf;
   private GoogleCredentials credentials;
   private static final Gson GSON = new Gson();
+  private static final Logger LOG = LoggerFactory.getLogger(ServiceAccountAccessTokenProvider.class);
+  public static final int DEFAULT_INITIAL_RETRY_DURATION_SECONDS = 5;
+  public static final int DEFAULT_MAX_RETRY_COUNT = 5;
+  public static final int DEFAULT_MAX_RETRY_DURATION_SECONDS = 80;
+
 
   @Override
   public AccessToken getAccessToken() {
-    try {
-      com.google.auth.oauth2.AccessToken token = getCredentials().getAccessToken();
-      if (token == null || token.getExpirationTime().before(Date.from(Instant.now()))) {
-        refresh();
-        token = getCredentials().getAccessToken();
+    int initialRetryDuration = DEFAULT_INITIAL_RETRY_DURATION_SECONDS;
+    int maxRetryCount = DEFAULT_MAX_RETRY_COUNT;
+    int maxRetryDuration =  DEFAULT_MAX_RETRY_DURATION_SECONDS;
```

Comment:
why are we creating copy of instance variables here? It eliminates the purpose of them being `final & static`